gnocchi/carbonara.py

```diff
@@ -157,6 +159,18 @@ def _scipy_aggregate(self, method, remove_unique=False, *args, **kwargs):
         timestamps = tstamps.astype('datetime64[ns]', copy=False)
         return pandas.Series(values, pandas.to_datetime(timestamps))
 
+    def derived(self):
+        timestamps = self._ts.index[1:]
+        # NOTE(sileht): we use absolute() as best-effort for cumulative counter
+        # reset, what ever the solution we use, the first point after the
+        # counter reset will always be wrong, because we can't known when
+        # the counter have been reseted between two points.
+        values = numpy.absolute(numpy.diff(self._ts.values))
```

Comment:
I'd say do not use absolute here. If you use absolute you lose information. It'd be better to store the relative diff, and then, with an API option, apply `abs` on values _if_ needed.